### PR TITLE
avocado.utils asset fetcher

### PR DESCRIPTION
**Pull request by apahim**
Utility to find for files in multiple locations, caching it when
successfully fetched.

Reference: https://trello.com/c/KTeMIx0u
Signed-off-by: Amador Pahim <apahim@redhat.com>

### CODE REVIEW

File: avocado/utils/asset.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    if expected_hash is None:
+        log.debug('Skipping hash check for %s.' % filename)
+        return True
+
+    file_hash = crypto.hash_file(filename)
+    if file_hash == expected_hash:
+        log.debug('Asset %s verified.' % filename)
+        return True
+    else:
+        log.debug('Asset %s corrupted.' % filename)
```
Comment (apahim) [inline avocado/utils/asset.py]: s/debug/error/

File: avocado/utils/asset.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+    :param params: yaml parsed options.
+    :param search: (optional) search for yaml option with all the asset
+            data. Default is 'asset'.
+    :return: asset path.
+    """
+
+    name = params.get('name', '*/%s/*' % search, default=None)
+    md5sum = params.get('md5sum', '*/%s/*' % search, default=None)
+    cache_dir = params.get('cache_dir', '*/%s/*' % search, default='/var/tmp')
```
Comment (lmr) [inline avocado/utils/asset.py]: I was taking a quick look at this, and it occurred to me that  some times we won't want to explicitly set a cache dir. In that case, I'd argue that it's best to create one with `tmpfile.mkdtemp`, for security purposes (avoid exploits of predictable tmp paths).

Comment (lmr) [inline avocado/utils/asset.py]: For the other situations where one wants to set the cache dir, it's advisable to not set the cache dir to a temporary directory (think iso files).

File: avocado/utils/asset.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+            log.debug('Trying to download from %s.' % url)
+            try:
+                download._get_file(url, dst)
+            except Exception as e:
+                log.error(e)
+                continue
+            if _check_file(dst, md5sum):
+                return dst
+
+        if urlobj.scheme == 'ftp':
```
Comment (apahim) [inline avocado/utils/asset.py]: s/if/elif/

File: avocado/utils/asset.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+            log.debug('Trying to download from %s.' % url)
+            try:
+                download._get_file(url, dst)
+            except Exception as e:
+                log.error(e)
+                continue
+            if _check_file(dst, md5sum):
+                return dst
+
+        if urlobj.scheme == 'file':
```
Comment (apahim) [inline avocado/utils/asset.py]: s/if/elif/

File: avocado/utils/asset.py (hunk 2 above)
Comment (apahim) [inline avocado/utils/asset.py]: Makes sense.

**Comment by apahim** [comment]
v2 coming soon.

**Pull request closed without merging**